templates/stretch/etc/fail2ban/jail.conf.erb

```diff
@@ -72,7 +72,7 @@ destemail = <%= scope['::fail2ban::email'] %>
 sendername = Fail2Ban
 
 # Email address of the sender
-sender = fail2ban@localhost
+sender = <%= scope['::fail2ban::sender'] %>
```

Comment:
can you please use `@fail2ban::sender` to access the variable?

Reply:
I think you meant `$fail2ban::sender`, but this is an erb template, not an epp one :smiley:

Reply:
The desired way to access a variable in an erb template is `@class::variable`:
https://puppet.com/docs/puppet/5.5/lang_template_erb.html#variable

Reply:
Just tested to be absolutely certain that it is not working as expected in erb.

`@fail2ban` is `nil`, and `nil::sender` is basically `nil.sender`, so:

```
fail2ban
  on debian-9-x86_64
    fail2ban::config
      when content template
        should contain File[fail2ban.conf] with ensure => "present", content =~ /THIS FILE IS MANAGED BY PUPPET/, notify => "Service[fail2ban]", require => "Package[fail2ban]" and content =~ /^chain = INPUT$/ (FAILED - 1)
      when content template (custom)
        should contain File[fail2ban.conf] with ensure => "present", content =~ /THIS FILE IS MANAGED BY PUPPET/, notify => "Service[fail2ban]", require => "Package[fail2ban]" and content =~ /^chain = INPUT$/ (FAILED - 2)
      when iptables chain provided
        should contain File[fail2ban.conf] with content =~ /^chain = TEST$/ (FAILED - 3)
      when custom banaction is provided
        should contain File[fail2ban.conf] with content =~ /^banaction = iptables$/ (FAILED - 4)
[...]
Failures:

  1) fail2ban on debian-9-x86_64 fail2ban::config when content template should contain File[fail2ban.conf] with ensure => "present", content =~ /THIS FILE IS MANAGED BY PUPPET/, notify => "Service[fail2ban]", require => "Package[fail2ban]" and content =~ /^chain = INPUT$/
     Failure/Error: return call_function('template', template_name) unless template_name.end_with?('.epp')
     
     Puppet::PreformattedError:
       Evaluation Error: Error while evaluating a Function Call, Failed to parse template fail2ban/stretch/etc/fail2ban/jail.conf.erb:
         Filepath: /usr/home/romain/Projects/puppet/modules/fail2ban/spec/fixtures/modules/fail2ban/templates/stretch/etc/fail2ban/jail.conf.erb
         Line: 75
         Detail: undefined method `sender' for nil:NilClass
```

Reply:
That maybe doesn't work because `sender` isn't a variable within the `fail2ban::jail` class. let's stick with the `scope` way.